packages/dataregistry-csvviewer-extension/src/datatypes.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license BSD-3-Clause
+ *
+ * Copyright (c) 2019 Project Jupyter Contributors.
+ * Distributed under the terms of the 3-Clause BSD License.
+ */
+
+import { DataTypeNoArgs, DataTypeStringArg } from "@jupyterlab/dataregistry";
+import { Widget } from "@phosphor/widgets";
+import { Observable } from "rxjs";
+
+// TODO: refactor
```

Comment:
This looks good to me at the moment. The datatypes will be properly typed, so if you do `datatypes.csv` it will know the type. What do you have in mind for refactor?

Reply:
I have the TODO here as this is a temporary hack until each of the converters is a separate extension. E.g., instead of L26, we'd import the data type from the widget data converter extension.